### ISSUE TITLE
Curl SSL certificate problem on Raspberry Pi 

### ISSUE THREAD

**Issue by dragosnicolae5555** [awslabs/amazon-kinesis-video-streams-producer-sdk-cpp#67]
./curl https://kinesisvideo.us-west-2.amazonaws.com --verbose      returns 
I have rebuild curl as instructed in  the  Troubleshooting section 
 
 CApath: none
* TLSv1.2 (OUT), TLS handshake, Client hello (1):
* TLSv1.2 (IN), TLS handshake, Server hello (2):
* TLSv1.2 (IN), TLS handshake, Certificate (11):
* TLSv1.2 (IN), TLS handshake, Server key exchange (12):
* TLSv1.2 (IN), TLS handshake, Server finished (14):
* TLSv1.2 (OUT), TLS handshake, Client key exchange (16):
* TLSv1.2 (OUT), TLS change cipher, Client hello (1):
* TLSv1.2 (OUT), TLS handshake, Finished (20):
* TLSv1.2 (IN), TLS handshake, Finished (20):
* SSL connection using TLSv1.2 / ECDHE-RSA-AES128-GCM-SHA256
* ALPN, server did not agree to a protocol
* Server certificate:
*  subject: CN=kinesisvideo.us-west-2.amazonaws.com
*  start date: Nov 21 00:00:00 2017 GMT
*  expire date: Nov 21 12:00:00 2018 GMT
*  subjectAltName: host "kinesisvideo.us-west-2.amazonaws.com" matched cert's "kinesisvideo.us-west-2.amazonaws.com"
*  issuer: C=US; O=Amazon; OU=Server CA 1B; CN=Amazon
*  SSL certificate verify ok.
> GET / HTTP/1.1
> Host: kinesisvideo.us-west-2.amazonaws.com
> User-Agent: curl/7.57.0
> Accept: */*
>
< HTTP/1.1 403 Forbidden
< x-amzn-RequestId: 57b17141-3b02-11e8-9c1b-dbec9b0dc602
< Content-Length: 127
< Date: Sun, 08 Apr 2018 07:56:27 GMT
<
<MissingAuthenticationTokenException>
  <Message>Missing Authentication Token</Message>
</MissingAuthenticationTokenException>
* C

**Comment by unicornss** [comment]
Could you pl. confirm if the file `cert.pem` exists in  `/etc/ssl/cert.pem`  - as mentioned in https://github.com/awslabs/amazon-kinesis-video-streams-producer-sdk-cpp#certificate-store-integration 

Thanks

**Comment by bdhandap** [comment]
@dragosnicolae5555 I see from the logs that you have posted, your curl CAPath is not set. Can you please follow the steps mentioned in [Troubleshooting section](https://github.com/awslabs/amazon-kinesis-video-streams-producer-sdk-cpp#curl-ssl-issue---unable-to-get-local-issuer-certificate) to reconfigure and build curl?

> ./curl https://kinesisvideo.us-west-2.amazonaws.com --verbose returns
I have rebuild curl as instructed in the Troubleshooting section
> 
> **CApath: none**

**Comment by briancarbs** [comment]
@dragosnicolae5555 I just finished compiling and running this demo on an rpi3 B+ with the latest raspbian build and ran into a similar issue. 

```
ERROR - curl perform failed for url https://kinesisvideo.us-east-1.amazonaws.com/describeStream with result Problem with the SSL CA cert (path? access rights?): error setting certificate verify locations:
```

I didn't explicitly set the CAPath but I did manually create a cert.pem file from scratch 
 in /etc/ssl/ and copied the root CA cert from [the link provided](https://www.amazontrust.com/repository/SFSRootCAG2.pem) the into that file. After that it worked fine for me. Hope this helps!
